src/main.rs

```diff
@@ -213,6 +214,109 @@ struct Args {
         env = "OTEL_EXPORTER_OTLP_ENDPOINT"
     )]
     otel_exporter_otlp_endpoint: Option<String>,
+    #[clap(long = "otel-metrics-endpoint", env = "OTEL_EXPORTER_METRICS_ENDPOINT", default_value = metrics::DEFAULT_OTEL_METRICS_ENDPOINT)]
+    otel_metrics_endpoint: String,
+    #[clap(long = "otel-metrics-mode", env = "OTEL_EXPORTER_METRICS_MODE", default_value_t = OtlpMode::Http)]
+    otel_metrics_mode: OtlpMode,
+    // Enable metrics
+    #[clap(
+        long = "enable-metrics",
+        default_value_t = false,
+        env = "WASMCLOUD_ENABLE_METRICS"
+    )]
+    enable_metrics: bool,
+    // Specify metrics exporter. Defaults to stdout but also supports otlp and prometheus
+    #[clap(
+        long = "metrics-type",
+        default_value_t = MetricType::Debug,
+        env = "WASMCLOUD_METRICS_TYPE"
+    )]
+    metrics_type: MetricType,
+    #[clap(long = "metrics-prometheus-port", env = "WASMCLOUD_METRICS_PROMETHEUS_PORT", default_value_t = metrics::DEFAULT_PROMETHEUS_PORT)]
+    metrics_prometheus_port: i32,
+}
+
+#[derive(clap::ValueEnum, Clone, Debug)]
+enum OtlpMode {
+    Otlp,
+    Http,
+}
+
+impl OtlpMode {
+    fn to_config(&self) -> metrics::OtlpMode {
+        match &self {
+            OtlpMode::Otlp => metrics::OtlpMode::Otlp,
+            OtlpMode::Http => metrics::OtlpMode::Http,
+        }
+    }
+}
+impl std::fmt::Display for OtlpMode {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let s = match self {
```

Comment:
could you get away with delegating harder by calling `to_config()` or `into()` here and then calling `.fmt(f)` ?